src/main/java/com/aws/greengrass/componentmanager/builtins/ArtifactDownloader.java

```diff
@@ -294,5 +295,12 @@ public boolean canSetFilePermissions() {
     public boolean canUnarchiveArtifact() {
         return true;
     }
+
+    /**
+     * Cleanup artifacts.
+     *
+     * @param componentStore componentStore
+     */
+    public abstract void cleanup(ComponentStore componentStore) throws Exception;
```

Comment:
I don't think this method should have any parameters. If the `ComponentStore` is needed to delete things, then that should be handled in the constructor of an instance of `ArtifactDownloader` instead.

This method should not throw a raw Exception. Create a new exception type, `PackageCleanupException` similar to `PackageDownloadException`